src/Domain/Amr/Helpers.cpp

```diff
@@ -145,6 +147,47 @@ std::vector<ElementId<VolumeDim>> ids_of_children(
   }
 }
 
+template <size_t VolumeDim>
+std::deque<ElementId<VolumeDim>> ids_of_joining_neighbors(
+    const Element<VolumeDim>& element,
+    const std::array<Flag, VolumeDim>& flags) {
+  using ::operator<<;
+  ASSERT(alg::count(flags, Flag::Join) > 0,
+         "Element " << element.id() << " is not joining given flags " << flags);
+  ASSERT(alg::count(flags, Flag::Split) == 0,
+         "Splitting and joining an Element is not supported");
+  std::deque<ElementId<VolumeDim>> result;
```

Comment:
Maybe do a double pass to be able to use `reserve` to avoid repeated allocations? Or possibly reserve the max size. I'm not sure how big the max size is and how long this would be kept around in terms of memory overhead.

Reply:
there is no `reserve` for a `std::deque`.   In 3D the container returned by this function will have 1-4 elements.  
The deque is not kept around long.  It will probably be more clear once I add the action that uses this function, so I could drop this commit for now if you like

Reply:
It's fine to keep. My concern is that, independent of the length of the deque, it might do a lot of allocations (e.g. if it grows to size 4 it might do as many as 4 allocations). But we can address that later, and maybe see if Boost has a static deque, or implement our own. We'll see :)

Reply:
> static deque

I've pretty much written this for my new time-stepper history implementation, although I haven't separated it out into a separate utility.  I'll see how easy it is to factor out, since it sounds like it might be useful elsewhere.